app/services/case_validate_rejected_offender_sar_service.rb

```diff
@@ -0,0 +1,31 @@
+class CaseValidateRejectedOffenderSARService
+  attr_reader :result, :error_message
+
+  def initialize(user:, kase:, params:)
+    @user = user
+    @case = kase
+    @params = params
+    @result = :incomplete
+    @error_message = nil
+  end
+
+  def call(message = nil)
+    ActiveRecord::Base.transaction do
+      @case.assign_attributes(@params)
+      @case.save # rubocop:disable Rails/SaveBang
+      @case.number.tr!("R", "")
```

Comment:
I think it would be useful to extract the number change into its own method for clarity.